internal/manager/validated.go

```diff
@@ -0,0 +1,48 @@
+package manager
+
+import "fmt"
+
+// ValidatedValue implements `pflag.Value` interface. It can be used for hooking up arbitrary validation logic to any type.
+// It should be passed to `pflag.FlagSet.Var()`.
+type ValidatedValue[T any] struct {
+	origin      string
+	variable    *T
+	constructor func(string) (T, error)
+}
+
+// NewValidatedValue creates a validated variable of type T. Constructor should validate the input and return an error
+// in case of any failures. If validation passes, constructor should return a value that's to be set in the variable.
```

Comment:
Either here on in the struct comment itself: mention that the string argument in the `constructor` will be what's passed in by the user at the command line (or env due to our mechanism of envs binding)